### ISSUE TITLE
Extract CTAs in Notification into its own component

### ISSUE THREAD

**Issue by aabenoja** [WTW-IM/es-components#73]
#72 introduces additional complexity to what call to action objects are accepted. CTA support should be ripped out of Notification into its own component in `Patterns`

**Comment by aabenoja** [comment]
Here's a rough idea of what I have in mind:

```js
<Notification {...notificationProps}>
  <p>Notification Content<p>
  <CallsToAction>
    <CallsToAction.Simple handleOnClick={handler}>Text</CallsToAction.Simple>
    <CallsToAction.Async handleOnClick={asyncHandler}>Text</CallsToAction.Async>
  </CallsToAction>
<Notification>

CallsToAction.Async = class AsyncAllToAction extends React.Component {
  constructor(props) {
    super(props);
    this.state = { isProcessing: false };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e) {
    const promise = this.props.handleOnClick(e);
    if (!_.isFunction(promise.then) || !_.isFunction(promise.catch)) {
      throw new Error('handleOnClick for CallsToAction.Async must return a promise');
    }

    this.setState({ isProcessing: true });
    promise.catch(() => console.error('An issue occurred with your request'))
      .then(() => this.setState({ isProcessing: false }));
  }

  render() {
    const { children, disabled, ...props } = this.props;
    const { isProcessing } = this.state;
    return (
      <Button
        {...props}
        disabled={disabled || isProcessing}
        handleOnClick={this.handleClick}>
        {children}
      </Button>
    );
  }
};
```

String to think that `AsyncButton` can be its own pattern.

**Comment by jrios** [comment]
The CallToAction component should encapsulate the Notification. This is what turns it into a pattern.

```javascript
class CallToAction extends React.Component {
  render() {
    return (
      <Notification>
         <CallToAction.Simple />
         <CallToAction.Async />
      </Notification>
    );
  }
}
```

**Comment by jrios** [comment]
I am taking this on as well as working to simplify the Notifications component itself.

**Comment by jrios** [comment]
closed with #236
